### PR TITLE
update to Julia v0.7, prepare for 1.0

### PR DESCRIPTION
**Pull request by RalphAS**
This builds on @Cody-G's work to qualify `axes` calls, and attempts to get the package working on Julia v0.7.  A new version of AxisArrays will presumably be needed for this to work on v1.0.

### CODE REVIEW

**Comment by timholy** [comment]
@mbaumann is apparently on it. Qualifying the calls is exactly the right
approach. The ones to Base must be too. (Sorry, on phone, can't really look
at code.)

**Comment by RalphAS** [comment]
In case it's not clear, here I am importing everything from AxisArrays **except** `axes`, using qualified `AxisArrays.axes` where appropriate, and using `Base.axes` without qualification.  This was done with staged commits to avoid confusion; the fact that Cody's PR works on v0.6 really helped.  I figured this would make things clean for the future.

**Comment by RalphAS** [comment]
Rebased on master. Tests pass on Julia v0.7, but can't run on v1.0 until AxisArrays has been overhauled.

**Comment by kmsquire** [comment]
@RalphAS, sorry, I should have looked here before merging #26.  Thanks for rebasing.

File: .travis.yml
Comment (timholy) [inline .travis.yml:18]: Is the deployment not working? There is a deploy key in the repo and two encrypted keys filed with Travis.

But of course we may not need the docs here, now that we have centralized docs with JuliaImages.

**Comment by timholy** [comment]
Let's get some new runs here now that ImageCore is tagged.

**Review by timholy — APPROVED** [review]
Looks great, thanks yet again!

We may not even need to wait for CI to pass, given that it's completely broken as-is.

File: .travis.yml
Comment (RalphAS) [reply .travis.yml:18]: No, I misunderstood where the deployment conditions are handled (in the docs/make.jl).  Let's fix that in the final round for v1.0.

**Pull request merged**
